mangaki/mangaki/fields.py

```diff
@@ -0,0 +1,56 @@
+from django.db import models
+
+from mangaki.choices import Category
+
+# See Django custom fields documentation at
+# https://docs.djangoproject.com/en/1.9/howto/custom-model-fields/j
+class CategoryField(models.IntegerField):
+    """
+    A custom field that handles automatically conversion between an integer
+    (stored in the database) and a convenient Category object.
+
+    This field handles conversion from integer, strings, and Category objects,
+    allowing the following kind of patterns:
+
+        Work.objects.filter(category=1)
+        Work.objects.filter(category='anime')
+        Work.objects.last().category.slug # 'anime', 'manga', etc.
+        str(Work.objects.last().category) # 'Anime', 'Manga', etc.
+    """
+
+    description = "A work category"
+
+    def __init__(self, *args, **kwargs):
+        kwargs['choices'] = Category.CHOICES.items()
+        super().__init__(*args, **kwargs)
+
+    def get_prep_value(self, value):
+        if isinstance(value, str):
+            value = Category(value)
+
+        return super().get_prep_value(value)
+
+    def get_prep_lookup(self, lookup_type, value):
+        if lookup_type == 'exact':
+            return self.get_prep_value(value)
+        elif lookup_type == 'in':
+            return [self.get_prep_value(v) for v in value]
+        else:
+            raise TypeError('Lookup type %r not supported.' % lookup_type)
+
+    def from_db_value(self, value, expression, connection, context):
+        if value is None:
+            return value
+
+        return Category(value)
+
+    def to_python(self, value):
+        if isinstance(value, Category):
+            return value
+
+        pyvalue = super().to_python(value)
+
+        if pyvalue is None:
+            return value
```

Comment:
Cf la [doc linkée en haut du fichier](https://docs.djangoproject.com/en/1.9/howto/custom-model-fields/#converting-values-to-python-objects). En résumé, cette fonction "Converts the value into the correct Python object. It acts as the reverse of value_to_string(), and is also called in clean()."

En vrai dans ce cas ci je pense que se baser sur le `super()` diminue la lisibilité plus qu'autre chose, je vais changer.